src/Diagnostics.ModelsAndUtils/Models/ResponseExtensions/Guage.cs

```diff
@@ -0,0 +1,231 @@
+        ///             guages.Add(g);             
+        ///         }
+        ///         else
+        ///         {
+        ///             if(util == 21)
+        ///             {
+        ///                 Guage g = new Guage(InsightStatus.Critical, util, "`" + util.ToString() + " %`", "`" + util.ToString() + "% Utilized`", GuageSize.Medium, "`Some markdown string`");
+        ///                 guages.Add(g);                
+        ///             }
+        ///             else
+        ///             {
+        ///                 Guage g = new Guage(InsightStatus.Info, util, "`" + util.ToString() + " %`", "`" + util.ToString() + "% Utilized`", GuageSize.Small, "`Some markdown string`");
+        ///                 guages.Add(g);
+        ///             }
+        ///         }
+        ///     }
+        ///     res.AddGuages(guages, GuageRenderDirection.Horizontal, GuageSize.Inherit);
+        ///     res.AddInsight(InsightStatus.Info, "More detailed info below");
+        ///     res.AddGuages(guages, GuageRenderDirection.Vertical, GuageSize.Large);
+        ///     return res;
+        /// }
+        /// </code>
+        /// </example>
+        public static DiagnosticData AddGuages(this Response response, List<Guage> guages, GuageRenderDirection renderDirection, GuageSize guageSize)
+        {
+            try
+            {
+                if (guages == null || !guages.Any())
+                    return null;
+
+                var table = new DataTable();
+                table.Columns.Add("RenderDirection", typeof(string));
+                table.Columns.Add("MasterSize", typeof(string));
+                table.Columns.Add("Size", typeof(string));
+                table.Columns.Add("FillColor", typeof(int));
+                table.Columns.Add("PercentFilled", typeof(double));
+                table.Columns.Add("DisplayValue", typeof(string));
+                table.Columns.Add("Label", typeof(string));
+                table.Columns.Add("Description", typeof(string));
+
+
+                foreach (Guage g in guages)
+                {
+                    table.Rows.Add(
+                        JsonConvert.SerializeObject(renderDirection),
+                        JsonConvert.SerializeObject(guageSize),
+                        JsonConvert.SerializeObject(g.Size),
+                        g.Status,
+                        g.PercentFilled,
+                        g.DisplayValue,
+                        g.Label,
+                        g.Description
+                        );
+
+                }
+
+                var diagData = new DiagnosticData()
+                {
+                    Table = table,
+                    RenderingProperties = new Rendering(RenderingType.Guage)
+                };
+                response.Dataset.Add(diagData);
+                return diagData;
+            }
+            catch (Exception ex)
```

Comment:
What a special catch if you are just throwing the exception

Reply:
I wasn't sure if the exception needs to be bubbled up and what the calling code is doing with it. I saw other extensions and they do not have a try in them. There is no specific action that I can take on an exception. Should I remove the try catch ?

Reply:
removed the try catch block